Makefile

```diff
@@ -98,6 +98,7 @@ e2e-test-docker-image: deploy/e2e-test-image/Dockerfile
 
 .PHONY: e2e-test
 e2e-test: deps ## Run e2e tests
+	@docker pull $(E2E_TEST_DOCKER_IMAGE)
```

Comment:
Good trick! I have now implemented it the Makefile-way, as I find that the individual steps are then more together. But if you think that Github can optimise better if the itself hosted images are in the YAML, or something else, then I will adapt it that way.

Reply:
That's true and I also like the close relationship between pulling the image and the usage in the tests 🙂